firewood/src/v2/api.rs

```diff
@@ -170,6 +171,41 @@ pub trait DbView {
         last_key: Option<K>,
         limit: Option<usize>,
     ) -> Result<Option<RangeProof<Vec<u8>, Vec<u8>>>, Error>;
+
+    /// Obtain a stream over the keys/values of this view, using an optional starting point
+    ///
+    /// # Arguments
+    ///
+    /// * `first_key` - If None, start at the lowest key
```

Comment:
I'm not sure why we need to provide the three APIs all together, why not just 1) one `iter_option` or 2)`iter` and `iter_from`. I would prefer the latter though.

Reply:
Richard previously pointed out that it would be better to do the latter, but rather than have each implementation have two new methods, you can only add one (the iter_option) and both automatically take the default implementation from the trait.